tools/clang/unittests/HLSL/ExecutionTest.cpp

```diff
@@ -4146,7 +4162,566 @@ TEST_F(ExecutionTest, ATOSampleCmpLevelTest) {
+    float r = (float)x;
+    float g = (float)y;
+    // provide some different values just to fill in b and a
+    float b = (float)(x + y)*0.5f;
+    float a = (float)(x + y)*0.1f;
+    RGBA[i].SetChannels(r, g, b, a);
+  }
+  virtual void *GetElements() { return (void*)RGBA; }
+  virtual unsigned GetXDim() { return xdim; }
+  virtual unsigned GetYDim() { return ydim; }
+  virtual DXGI_FORMAT GetFormat() override { return m_format; };
+};
+
+template <unsigned xdim, unsigned ydim>
+struct RawFloatR11G11B10ATexture : public ExecutionTest::RawGatherTexture {
+  FloatR11G11B10 RGBA[xdim*ydim];
+  // Set i'th element to floatified x,y and some derived values
+  virtual void SetElement(int i, int x, int y) override {
+    float r = (float)x;
+    float g = (float)y;
+    float b = (float)(x + y)*0.5f;
+    RGBA[i].SetChannels(r, g, b, 0);
+  }
+  virtual void *GetElements() { return (void*)RGBA; }
+  virtual unsigned GetXDim() { return xdim; }
+  virtual unsigned GetYDim() { return ydim; }
+  virtual DXGI_FORMAT GetFormat() override { return DXGI_FORMAT_R11G11B10_FLOAT; };
+};
+
+template <typename RGBAType, unsigned xdim, unsigned ydim>
+struct RawIntTexture : public ExecutionTest::RawGatherTexture {
+  bool m_isSigned;
+  bool m_isNorm;
+  unsigned m_maxVal;
+  DXGI_FORMAT m_format;
+  RGBAType RGBA[xdim*ydim];
+  // Increase maxdim by 2 to make some more interesting values for B and A
+  RawIntTexture(bool isSigned, bool isNorm, int maxVal, DXGI_FORMAT format)
+    : m_isSigned(isSigned), m_isNorm(isNorm), m_maxVal(maxVal + 2), m_format(format) {
+    if (isSigned)
+      m_maxVal /= 2;
+  }
+  // Set i'th element to values scaled per max dimentions for norms, shifted for signed
+  // but otherwise just the x and y values themselves
+  virtual void SetElement(int i, int x, int y) override {
+    double fr = x;
+    double fg = y;
+    // provide some different values just to fill in b and a
+    double fb = x + 2;
+    double fa = y + 2;
+    // If signed, get some unsigned values in there
+    if (m_isSigned) {
+      fr -= m_maxVal;
+      fg -= m_maxVal;
+      fb -= m_maxVal;
+      fa -= m_maxVal;
+    }
+    // If normalized, scale to given range
+    if (m_isNorm) {
+      fr /= m_maxVal;
+      fg /= m_maxVal;
+      fb /= m_maxVal;
+      fa /= m_maxVal;
+
+      fr *= (1 << (RGBAType::GetRSize() - m_isSigned - 1));
+      fg *= (1 << (RGBAType::GetGSize() - m_isSigned - 1));
+      fb *= (1 << (RGBAType::GetBSize() - m_isSigned - 1));
+      fa *= (1 << (RGBAType::GetASize() - m_isSigned - 1));
+    }
+    RGBA[i].SetChannels((UINT)fr, (UINT)fg, (UINT)fb, (UINT)fa);
+  }
+  virtual void *GetElements() { return (void*)RGBA; }
+  virtual unsigned GetXDim() { return xdim; }
+  virtual unsigned GetYDim() { return ydim; }
+  virtual DXGI_FORMAT GetFormat() override { return m_format; };
+};
+
+#define RAWGATHER_FALLBACK // Enable to use pre-6.7 fallback mechanisms to vet raw gather tests
```

Comment:
Fallback worked for me with the following changes, but I don't think you want to commit with fallback enabled.
```
diff --git a/tools/clang/unittests/HLSL/ExecutionTest.cpp b/tools/clang/unittests/HLSL/ExecutionTest.cpp
index ee61606ff..18defc085 100644
--- a/tools/clang/unittests/HLSL/ExecutionTest.cpp
+++ b/tools/clang/unittests/HLSL/ExecutionTest.cpp
@@ -4488,13 +4488,14 @@ void ExecutionTest::DoRawGatherTest(ID3D12Device *pDevice, RawGatherTexture *raw

   // Create shaders
 #ifdef RAWGATHER_FALLBACK
-  const wchar_t *target = L"cs_6_6";
+  const wchar_t *target = L"cs_6_2";
 #else
   const wchar_t *target = L"cs_6_7";
 #endif
+  const wchar_t *options[] = {L"-enable-16bit-types"};

   CComPtr<ID3D12PipelineState> pPSO;
-  CreateComputePSO(pDevice, pRootSignature, pShader, target, &pPSO);
+  CreateComputePSO(pDevice, pRootSignature, pShader, target, &pPSO, options, _countof(options));

   // Reset commandlist to shader PSO
   VERIFY_SUCCEEDED(pCommandList->Reset(pCommandAllocator, pPSO));
@@ -4580,17 +4581,20 @@ TEST_F(ExecutionTest, ATORawGather) {

   CComPtr<ID3D12Device> pDevice;
 #ifdef RAWGATHER_FALLBACK
-  D3D_SHADER_MODEL sm = D3D_SHADER_MODEL_6_6;
+  D3D_SHADER_MODEL sm = D3D_SHADER_MODEL_6_2;
 #else
   D3D_SHADER_MODEL sm = D3D_SHADER_MODEL_6_7;
 #endif
   if (!CreateDevice(&pDevice, sm))
       return;

-  if (DoesDeviceSupportAdvancedTexOps(pDevice)) {
+#ifndef RAWGATHER_FALLBACK
+  if (!DoesDeviceSupportAdvancedTexOps(pDevice)) {
     WEX::Logging::Log::Comment(L"Device does not support Advanced Texture Operations.");
     WEX::Logging::Log::Result(WEX::Logging::TestResults::Skipped);
+    return;
   }
+#endif

   static const int NumThreadsX = 32;
   static const int NumThreadsY = 32;
```